### PR TITLE
[SD-1377] Display addresses from countries supported in the current store

### PR DESCRIPTION
**Pull request by v10110**
https://spark-solutions.atlassian.net/browse/SD-1377

### CODE REVIEW

**Review by damianlegawiec — APPROVED** [review]

File: api/spec/requests/spree/api/v2/storefront/account/addresses_spec.rb [hunk 1]
```diff
@@ -67,6 +67,23 @@
       end
     end
 
+    context 'when address from countries that are supported in the current store' do
+      let!(:country) { create(:country, name: 'France') }
+      let!(:zone) { create(:zone, name: 'EU_VAT', countries: [country], kind: 'country') }
+      let!(:eu_address) { create(:address, user_id: user.id, country: country) }
+
+      before do
+        Spree::Config[:checkout_zone] = zone.name
```
Comment (damianlegawiec) [inline api/spec/requests/spree/api/v2/storefront/account/addresses_spec.rb:76]: @v10110 please do not use this preference, please use `current_store.checkout_zone` instead

**Review by damianlegawiec — CHANGES_REQUESTED** [review]

File: api/spec/requests/spree/api/v2/storefront/account/addresses_spec.rb [hunk 2]
```diff
@@ -67,6 +67,24 @@
...
     end
 
+    context 'when address from countries that are supported in the current store' do
+      let(:current_store) { Spree::Store.first }
+      let!(:country) { create(:country, name: 'France') }
+      let!(:zone) { create(:zone, name: 'EU_VAT', countries: [country], kind: 'country') }
+      let!(:eu_address) { create(:address, user_id: user.id, country: country) }
+
+      before do
+        current_store.update(checkout_zone: zone)
```
Comment (damianlegawiec) [inline api/spec/requests/spree/api/v2/storefront/account/addresses_spec.rb:77]:
```suggestion
        store.update(checkout_zone: zone)
```

File: api/spec/requests/spree/api/v2/storefront/account/addresses_spec.rb [hunk 3]
```diff
@@ -67,6 +67,24 @@
       end
     end
 
+    context 'when address from countries that are supported in the current store' do
+      let(:current_store) { Spree::Store.first }
```
Comment (damianlegawiec) [inline api/spec/requests/spree/api/v2/storefront/account/addresses_spec.rb:71]:
```suggestion
```

Comment (v10110) [reply api/spec/requests/spree/api/v2/storefront/account/addresses_spec.rb:71]: @damianlegawiec we don't have store variable here, so I moved it to the top of the file

File: api/spec/requests/spree/api/v2/storefront/account/addresses_spec.rb [hunk 4]
```diff
@@ -67,6 +68,23 @@
       end
     end
 
+    context 'when address from countries that are supported in the current store' do
+      let!(:country) { create(:country, name: 'France') }
+      let!(:zone) { create(:zone, name: 'EU_VAT', countries: [country], kind: 'country') }
+      let!(:eu_address) { create(:address, user_id: user.id, country: country) }
```
Comment (damianlegawiec) [inline api/spec/requests/spree/api/v2/storefront/account/addresses_spec.rb:74]: @v10110 let's also create another address for this user from a country that isn't supported by this store to test if it doesn't show up

File: api/spec/requests/spree/api/v2/storefront/account/addresses_spec.rb [hunk 5]
```diff
@@ -6,6 +6,7 @@
   let!(:addresses) { create_list(:address, 3, user_id: user.id) }
   let!(:country) { create(:country, iso: 'GBR') }
   let!(:state) { create(:state, country: country) }
+  let(:store) { Spree::Store.first }
```
Comment (damianlegawiec) [inline api/spec/requests/spree/api/v2/storefront/account/addresses_spec.rb:9]:
```suggestion
  let(:store) { Spree::Store.default }
```

**Review by damianlegawiec — APPROVED** [review]

**Pull request merged**